### PR TITLE
build(deps): remove struts dependency

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

Struts was really only used by the old IntegrationTester (which has since been re-written with scala/ps and now located at `autotest/IntegTester`). The only reason for the remaining dependency was to support a unit test which was using a RequestServlet mock provided by struts.

Better to do this with Mockito instead, and remove the dependency and the old (no longer built) code.

(I did also note that when we run `./sbt test` as part of our build it also unfortunately builds the front-end. Need to look at improving that SBT dependency setup... But that's unrelated to this PR... Also running that locally didn't show me the test results, however it seems to work fine on GHA so will check on this PR.)

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/test/java/com/tle/com/tle/integration/lti/generic/GenericLtiWrapperExtensionTest.java [hunk 1]
```diff
@@ -4,15 +4,22 @@
...
-import org.apache.struts.mock.MockHttpServletRequest;
+import javax.servlet.http.HttpServletRequest;
 import org.junit.Test;
 
 @SuppressWarnings("nls")
 public class GenericLtiWrapperExtensionTest {
+  private static final String USERID_CUSTOM = "custom-user-id-val";
+  private static final String USERID_DEFAULT = "default-user-id-val";
+  private static final String USERNAME_CUSTOM = "custom-username-val";
+  private static final String USERNAME_DEFAULT = "default-username-val";
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/test/java/com/tle/com/tle/integration/lti/generic/GenericLtiWrapperExtensionTest.java:21]: Some general tidy up too.

File: Source/Plugins/Core/com.equella.core/test/java/com/tle/com/tle/integration/lti/generic/GenericLtiWrapperExtensionTest.java [hunk 2]
```diff
@@ -175,39 +182,40 @@ public void testIsPrefixUserIdFalse() {
     assertFalse(b);
   }
 
-  private MockHttpServletRequest setupMockRequestAllValues() {
-    MockHttpServletRequest req = new MockHttpServletRequest();
-    req.addParameter("mycustomid", "custom-user-id-val");
-    req.addParameter("mycustomname", "custom-username-val");
-    req.addParameter(ExternalToolConstants.LIS_PERSON_SOURCEDID, "default-username-val");
-    req.addParameter(ExternalToolConstants.USER_ID, "default-user-id-val");
+  private HttpServletRequest setupMockRequestNoValues() {
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/test/java/com/tle/com/tle/integration/lti/generic/GenericLtiWrapperExtensionTest.java:185]: I also swapped the order of the functions around to for readability.

**Comment by edalex-ian** [comment]
Hold on a minute... Struts is not used by the IntegTester, it's used by the IntegrationTester at `Source/Tools/IntegrationTester`.

But is IntegTester just a new version of that (done with Scala and PS)? We have no build for IntegrationTester (just a plugin-jpf.xml).....

**Comment by edalex-ian** [comment]
Yeah okay, IntegTester is most of IntegationTester but then also with CloudProvider test endpoints.

So... The IntegrationTester folder should be deleted, and then the struts dependencies can be completely removed.

Will look at tomorrow.

(Side note, had to restart CodeBuild as ScalaCheck tests failed with Selenium unable to connect......)

**Comment by edalex-ian** [comment]
Okay, now it should be ready to review @PenghaiZhang :wink:

File: autotest/IntegTester/build.sbt [hunk 3]
```diff
@@ -1,7 +1,5 @@
-import sbt.IO
 import Path.rebase
-
-import scala.sys.process.Process
```
Comment (PenghaiZhang) [inline autotest/IntegTester/build.sbt:4]: Did you also find this import useless ?

**Review by PenghaiZhang — APPROVED** [review]
Definitely LGTM!

File: autotest/IntegTester/build.sbt (hunk 3 above)
Comment (edalex-ian) [reply autotest/IntegTester/build.sbt:4]: Yup, IntelliJ did - and tidied it up with I mistakenly added the struts dep into here - so I thought I'd keep it. :grin:

**Pull request merged**